07_leukemia_patient/workflow/workflow.sh

```diff
@@ -0,0 +1,232 @@
+done
+
+for i in $(cat BCR_human.gtf) ; do
+    samtools faidx $TRANSCRIPTOME "$i"  >> BCR_human.fa
+done
+
+cat "$CUSTOM_FA" ABL1_human.fa BCR_human.fa > combined.fa
+cat "$CUSTOM_GTF" ABL1_human.gtf BCR_human.gtf > combined.gtf
+
+# index
+
+~/leukemia_bwamem2/bwa-mem2-2.2.1_x64-linux/bwa-mem2 index -p indexed combined.fa
+
+echo "BWAMEM2 finished"
+
+# run
+
+mkdir -p $WD/bwa_mem2/output
+cd $WD/bwa_mem2/output
+
+~/leukemia_bwamem2/bwa-mem2-2.2.1_x64-linux/bwa-mem2 mem $WD/bwa_mem2/genome/indexed $WD/sorted_duplicate_r2.fastq.gz -t 20 -k 80 -w 12 | samtools view -bS | samtools sort -o bwa_mem2.sorted.bam 
+
+# run STAR fusion --> the genome takes too long to generate (3 days), use premade genome
+
+mkdir -p $WD/star_fusion/output
+cd $WD/star_fusion/output
+
+singularity exec -e -B `pwd` -B $PATH_STAR_FUSION \
+       $PATH_SIMG_FILE \
+       STAR-Fusion \
+       --left_fq $WD/sorted_duplicate_r2.fastq.gz \
+       --genome_lib_dir $STAR_FUSION_GENOME \
+       --output_dir $WD/star_fusion/output > log.txt
+
+echo "STAR_fusion finished"
+
+# adding the CB and UB tag to the bwa mem2 bam file and generating count table
+
+cd $WD/bwa_mem2/output
+
+samtools view -b -F 4 bwa_mem2.sorted.bam > mapped.bam
+
+# removed unmapped reads
+
+samtools view -H mapped.bam > header.txt
+samtools view mapped.bam | awk 'BEGIN {OFS="\t"} {split($1, a, /[;,]/); print a[1], $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13, $14, $15, a[2], a[3]}' | cat header.txt - | samtools view -Sbh > annotated_bwa_mem2.sorted.bam
+
+rm header.txt mapped.bam
+
+# UMI deduplication based on CB and UMI and gene name field
+
+echo "# non deduplicated reads bwa mem2"
+
+samtools view annotated_bwa_mem2.sorted.bam | cut -f 1 | wc -l
+
+# bwa mem2 does not require deduplication as the reads are already primary alignments but running anyway
+
+samtools view annotated_bwa_mem2.sorted.bam | awk 'BEGIN{OFS="\t"}{print $1,$3,$16,$17}' | sort -k3,4 | uniq | cut -f1 > dedup_read_names.txt
+samtools view -H annotated_bwa_mem2.sorted.bam > header.txt
+samtools view annotated_bwa_mem2.sorted.bam | grep -Ff dedup_read_names.txt | cat header.txt - | samtools view -Sbh > deduplicated_annotated_bwa_mem2.sorted.bam
+
+echo "# deduplicated reads bwa mem2"
+wc -l dedup_read_names.txt
+
+#rm dedup_read_names.txt header.txt annotated_bwa_mem2.sorted.bam
+
+# count table for bwa_mem2 --> based on counting the occurences for each gene, cell and count. The .fastqs are already deduplicated so don't need to repeat. 
+# column 1: gene id, column 2: barcode id, column 3: counts
+# only count the primary alignment, even if multimapped --> one alignment for mapped read
+
+mkdir -p $WD/count_tables/bwamem2
+cd $WD/count_tables/bwamem2
+
+echo -e "Counts\tGene_id\tBarcode" > counts_bwa_mem2.txt
+
+samtools view $WD/bwa_mem2/output/deduplicated_annotated_bwa_mem2.sorted.bam | cut -f3,16 | sort | uniq -c >> counts_bwa_mem2.txt
+
+# do the same for STAR fusion based on the Chimeric.out.junction
```

Comment:
caution the same algo cannot work for starfusion, there one himeric alignment spans two bam lines/records (*caution*)